app-backend/tile/src/main/scala/routes/ToolRoutes.scala

```diff
@@ -133,26 +138,36 @@ class ToolRoutes(implicit val database: Database) extends Authentication
             complete {
               val nodeId = node.map(UUID.fromString(_))
               val colorRamp = providedRamps.get(colorRampName).getOrElse(providedRamps("viridis"))
-              val responsePng = for {
-                (toolRun, tool) <- LayerCache.toolAndToolRun(toolRunId, user)
-                (ast, params)   <- LayerCache.toolEvalRequirements(toolRunId, nodeId, user)
-                tile            <- OptionT({
-                  val tms = Interpreter.interpretTMS(ast, params.sources, params.overrides, source, 256)
-                  logger.debug(s"Attempting to retrieve TMS tile at $z/$x/$y")
-                  tms(z, x, y).map {
-                    case Valid(op) => op.evaluateDouble
-                    case Invalid(errors) => throw InterpreterException(errors)
-                  }
-                })
-                cMap            <- LayerCache.toolRunColorMap(toolRunId, nodeId, user, colorRamp, colorRampName)
+              val responsePng: OptionT[Future, Png] = for {
+                ast   <- LayerCache.toolEvalRequirements(toolRunId, nodeId, user)
+                tile  <- OptionT({
+                        val literalAst: Future[Interpreted[MapAlgebraAST]] = BufferingInterpreter.literalize(ast, source, z, x, y)
+                        val futureTile: Future[Interpreted[Tile]] = literalAst.map({ validatedAst =>
+                          validatedAst.andThen({ resolvedAst =>
+                            logger.debug(s"Attempting to retrieve TMS tile at $z/$x/$y")
+                            BufferingInterpreter.interpret(resolvedAst, 256)(z, x, y).andThen({ lztile =>
+                              lztile.evaluateDouble match {
+                                case Some(t) =>
+                                  Valid(t)
+                                case None =>
+                                  Invalid(NEL.of(LazyTileEvaluationError(ast)))
+                              }
+                            })
+                          })
+                        })
+                        futureTile.map({ validatedTile =>
+                          validatedTile match {
+                            case Valid(t) => Option(t)
+                            case Invalid(errors) => throw InterpreterException(errors)
+                          }
+                        })
+                      })
+                cMap  <- LayerCache.toolRunColorMap(toolRunId, nodeId, user, colorRamp, colorRampName)
               } yield {
                 logger.debug(s"Tile successfully produced at $z/$x/$y")
-                params.metadata(ast.id).renderDef match {
-                  case Some(renderDef) =>
-                    tile.renderPng(renderDef)
-                  case None =>
-                    tile.renderPng(cMap)
-                }
+                ast.metadata.flatMap({ md =>
+                  md.renderDef.map({ renderDef => tile.renderPng(renderDef) })
+                }).getOrElse(tile.renderPng(cMap))
```

Comment:
Not sure if it's from here or elsewhere, but if a tile is requested outside the bounds of a project, this returns a `400` with a JSON error message, when it should probably return an empty PNG with a `200` or perhaps no PNG and a `204`.